example/src/example.erl

```diff
@@ -0,0 +1,64 @@
+-module(example).
+
+-export([start/0]).
+-export([start/2]).
+-export([stop/0]).
+-export([stop/1]).
+
+%% application
+%% @doc Starts the application
+start() ->
+  {ok, _Started} = application:ensure_all_started(example).
+
+%% @doc Stops the application
+stop() ->
+  application:stop(example).
+
+%% behaviour
+%% @private
+start(_StartType, _StartArgs) ->
+  {ok, Pid} = example_sup:start_link(),
+  {ok, _Pid} = start_listeners(),
+  {ok, Pid}.
+
+%% @private
+stop(_State) ->
+  ok = cowboy:stop_listener(example_http),
+  ok.
+
+start_listeners() ->
+  {ok, Port} = application:get_env(example, http_port),
+  {ok, ListenerCount} = application:get_env(example, http_listener_count),
+
+  RootTrail = trails:trail(<<"/">>, example_root_handler, [],  #{get => #{}}),
+  DescriptionTrail =
+    trails:trail(<<"/description">>
+                , example_description_handler
+                , []
+                ,  #{get => #{}}),
+  MessageTrail = trails:trail(<<"/message/[:echo]">>
```

Comment:
use `trails:trails([example_echo_handler])`, instead of building the routes here.